src/plugins/syscalls/win.cpp

```diff
@@ -353,17 +355,17 @@ static bool trap_syscall_table_entries(drakvuf_t drakvuf, vmi_instance_t vmi, sy
 
 void setup_windows(drakvuf_t drakvuf, syscalls *s)
 {
-    vmi_lock_guard lg(drakvuf);
+    auto vmi = vmi_lock_guard(drakvuf);
 
     addr_t start = 0;
 
-    if ( s->pm == VMI_PM_IA32E )
+    if ( !s->is32bit )
     {
         if ( !drakvuf_get_kernel_symbol_rva(drakvuf, "KiSystemServiceStart", &start) )
             throw -1;
 
         system_service_table_x64 sst[2] = {};
-        if ( VMI_FAILURE == vmi_read_ksym(lg.vmi, "KeServiceDescriptorTableShadow", 2*sizeof(system_service_table_x64), (void*)&sst, NULL) )
+        if ( VMI_FAILURE == vmi_read_ksym(vmi, "KeServiceDescriptorTableShadow", 2*sizeof(system_service_table_x64), (void*)&sst, NULL) )
```

Comment:
I'm surprised this works with `auto vmi`

Reply:
This is implicit cast operator magic of C++.

Reply:
Very handy